pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/route/CreateTemporaryUserRoute.java

```diff
@@ -77,9 +77,10 @@ public Object handle(Request request, Response response, CreateTemporaryUserPayl
             try {
                 User user = handle.attach(UserDao.class).createTempUser(clientDto.getId());
                 //create profile
-                UserProfileDao profileDao = handle.attach(UserProfileDao.class);
-                profileDao.createProfile(new UserProfile.Builder(user.getId())
-                        .setPreferredLangId(languageId)
+                handle.attach(UserProfileDao.class).createProfile(UserProfile.builder()
+                        .userId(user.getId())
+                        .preferredLangId(languageId)
+                        .preferredLangCode(null)
```

Comment:
I was thinking about it. And since we're not using custom builder anymore, I guess the answer is no. But on the other side, if we compare the huge builder class that can be replaced with the one annotation and some places with two calls, I would vote for the 2nd option.

Reply:
That would be fine if these two value were not related. Being able to set both allows the profile to be put into an invalid state.